examples/cmri_mpca/main.py

```diff
@@ -70,43 +70,69 @@ def main():
-            marker_locs=landmarks,
-            im_kwargs=dict(cfg.IM_KWARGS),
-            marker_kwargs=dict(cfg.MARKER_KWARGS),
-            n_cols=10,
-        ).savefig(str(save_images_location) + "/0)first_phase.png")
+        n_img_per_fig = 35
+        n_figures = int(n_samples / n_img_per_fig) + 1
+        for k in range(n_figures):
+            visualize.plot_multi_images(
+                [images[i][0, ...] for i in range(k * n_img_per_fig, min((k + 1) * n_img_per_fig, n_samples))],
+                marker_locs=landmarks[k * n_img_per_fig : min((k + 1) * n_img_per_fig, n_samples), :],
+                im_kwargs=dict(cfg.IM_KWARGS),
+                marker_cmap="Set1",
+                marker_kwargs=dict(cfg.MARKER_KWARGS),
+                marker_titles=markers,
+                image_titles=list(patient_ids[k * n_img_per_fig : min((k + 1) * n_img_per_fig, n_samples)]),
+                n_cols=5,
+            ).savefig(str(save_images_location) + "/0)landmark_visualization_%s_of_%s.png" % (k + 1, n_figures))
 
     # ---- data pre-processing ----
     # ----- image registration -----
     img_reg, max_dist = reg_img_stack(images.copy(), landmarks, landmarks[0])
     if save_images:
         visualize.plot_multi_images(
-            [img_reg[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_reg[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
+            image_titles=list(patient_ids),
         ).savefig(str(save_images_location) + "/1)image_registration")
 
     # ----- masking -----
     img_masked = mask_img_stack(img_reg.copy(), mask)
     if save_images:
         visualize.plot_multi_images(
-            [img_masked[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_masked[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
+            image_titles=list(patient_ids),
         ).savefig(str(save_images_location) + "/2)masking")
 
     # ----- resize -----
     img_rescaled = rescale_img_stack(img_masked.copy(), scale=1 / cfg.PROC.SCALE)
     if save_images:
         visualize.plot_multi_images(
-            [img_rescaled[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_rescaled[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
+            image_titles=list(patient_ids),
         ).savefig(str(save_images_location) + "/3)resize")
 
     # ----- normalization -----
     img_norm = normalize_img_stack(img_rescaled.copy())
     if save_images:
         visualize.plot_multi_images(
-            [img_norm[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_norm[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
```

Comment:
If you are using the same para multiple times, config seems better in the long term.